File: leetsync/2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.java
class Solution {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int m = nums1.length;
        int n = nums2.length;
        int result = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums1) {
            map.put(num, map.getOrDefault(num, 0) + n);
        }
        for (int num : nums2) {
            map.put(num, map.getOrDefault(num, 0) + m);
        }
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            int key = entry.getKey();
            int value = entry.getValue();
            if (value % 2 == 1) {
                result ^= key;
            }
        }
        return result;
    }
}

File: leetsync/2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.go
func xorAllNums(nums1 []int, nums2 []int) int {
	m := len(nums1)
	n := len(nums2)
	result := 0
	if m%2 == 1 {
		for _, v := range nums2 {
			result = result ^ v
		}

	}
	if n%2 == 1 {
		for _, v := range nums1 {
			result = result ^ v
		}
	}
	return result
}